chore(test):  Code Coverage must ignore some `matrix-sdk-crypto-(js|nodejs)`

Hywan opened the pull request:
This patch fixes https://github.com/matrix-org/matrix-rust-sdk/issues/748 and https://github.com/matrix-org/matrix-rust-sdk/issues/749.

The first patch uses [the Ignoring Paths feature](https://docs.codecov.com/docs/ignoring-paths) of codecov.io. I hope it will exclude entirely the `crates/matrix-sdk-crypto-(js|nodejs)/` paths from their UI and code coverage score.

The second patch updates the `tarpaulin` configuration file to:
1. fix the `exclude-files` paths (because they were invalid),
2. `exclude` crates entirely
3. be used in the CI directly, so that we don't have different configurations.

poljar commented on tarpaulin.toml:
```diff
@@ -1,4 +1,5 @@
 verbose = false
-exclude-files = ["matrix_sdk/examples/*", "matrix_sdk_common", "matrix_sdk_test"]
+exclude-files = ["crates/matrix_sdk/examples/*", "crates/matrix_sdk_common", "crates/matrix_sdk_test"]
+exclude = ["matrix-sdk-crypto-js", "matrix-sdk-crypto-nodejs"]
```
Why not add the `common` and `test` crate here as well if the `exclude` parameter is smart enough to handle crates?

We should also add the rest of the bindings crates and the `matrix-sdk-indexeddb` crate as well. Furthermore the `xtask` setup and the `labs` should be ignored as well.

![image](https://user-images.githubusercontent.com/552026/173319727-b9f25b74-72ab-44c6-bfa9-0cc9e16a9ef5.png)

poljar reviewed (changes requested):
I think this needs a bit more twiddling but thanks for working on this.

Hywan commented:
I've updated `tarpaulin` to exclude `matrix-sdk-test`, `matrix-sdk-indexeddb`, `matrix-sdk-crypto-ffi` in addition to `…-crypto-js` and `…-crypto-nodejs`.

I've updated codecov.io to ignore `crates/matrix-sdk-crypto-(ffi|nodejs|js)`, in addition to `labs` and `xtask`.

poljar commented:
> I've updated `tarpaulin` to exclude `matrix-sdk-test`, `matrix-sdk-indexeddb`, `matrix-sdk-crypto-ffi` in addition to `…-crypto-js` and `…-crypto-nodejs`.
> 
> I've updated codecov.io to ignore `crates/matrix-sdk-crypto-(ffi|nodejs|js)`, in addition to `labs` and `xtask`.

`matrix-sdk-ffi` is another bindings crate, let's also add `matrix-sdk-test-macros` even though it seems to be already excluded somehow.

poljar reviewed (approved):
Looks sensible, please just add the remaining missing crates and feel free to merge.
